vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+adtr_sum <- adtr_sum %>%
+  mutate(
+    PDFL = if_else(PCHGNAD >= 20 & CHGNAD >= 5 | is.na(CRFL) & CRNFL == "Y", "Y", NA_character_)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, CHGNAD, PCHGNAD, PDFL)
+)
+```  
+
+## Derive Analysis Flags `ANL02FL`, `ANL03FL`, `ANL04FL` {#analflag}
+
+Based on your analysis, additional analysis flags could be set up in `ADTR`.
+Below are some examples
+
+Analysis 02 Flag only includes the visit with the lowest NADIR (lowest sum of
+diameter) compared across all visits.  Use PCHGNNAD or PCHG?
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      new_var = ANL02FL,
+      order = vars(AVAL),
+      mode = "first"
+    ),
+    filter = ANL01FL == "Y"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, ANL02FL)
+)
+```  
+
+Analysis 03 flag includes `ANL01FL="Y"` and the target lesion sum of diameter
+until patient has had PD. Use the `PDFL` variable.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  restrict_derivation(
+    derivation = derive_var_relative_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      new_var = ANL03FL,
+      condition = PDFL == "Y",
+      order = vars(ADY),
+      mode = "first",
+      selection = "before",
+      inclusive = TRUE
+    ),
+    filter = ANL01FL == "Y"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, PDFL, ANL03FL)
+)
+```  
+
+Analysis 04 flag includes all sum of diameters even where lesion identifiers
+match and unmatch if the patient has had PD. Use `CRFL`, `CRNFL`, `PDFL` variables.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  mutate(
+    ANL04FL = if_else(ANL01FL == "Y" | PDFL == "Y", "Y", NA_character_)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, PDFL, ANL01FL, ANL04FL)
+)
+```  
+
+
+```{r}
+adtr <- bind_rows(adtr, adtr_sum)
+```
+
+## Derive Analysis Sequence Number (`ASEQ`) {#aseq}
+
+The `{admiral}` function `admiral::derive_var_obs_number()` can be used to
+derive `ASEQ`:
+
+```{r eval=TRUE}
+adtr <- adtr %>%
+  derive_var_obs_number(
+    by_vars = vars(STUDYID, USUBJID),
+    order = vars(PARAMCD, AVISITN, TRSEQ),
+    check_type = "error"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr, USUBJID, ASEQ),
+  display_vars = vars(USUBJID, PARAMCD, AVISIT, ASEQ, AVAL)
+)
+```  
+
+
+## Add ADSL Variables {#adslvars}
+  
+If needed, the other `ADSL` variables can now be added. List of `ADSL`
+variables already merged held in vector `adsl_vars`.
+
+```{r eval=TRUE}
+adtr <- adtr %>%
+  derive_vars_merged(
+    dataset_add = select(adsl, !!!negate_vars(adsl_vars)),
```

Comment:
The adsl_vars object still only contains RANDDT, as in line 149:
`adsl_vars <- vars(RANDDT)`

Should we update this to include other columns from adsl, e.g. SEX, DOB, TRTSTDT

Reply:
The call adds all ADSL variables to `adtr`. `RANDDT` is excluded because it is already in `adtr`.